modules/order/tests/src/Kernel/OrderRefreshTest.php

```diff
@@ -0,0 +1,267 @@
+
+    /** @var \Drupal\commerce_order\Entity\Order $order */
+    $order = Order::create([
+      'type' => 'default',
+      'state' => 'draft',
+      'mail' => $this->user->getEmail(),
+      'uid' => 0,
+      'ip_address' => '127.0.0.1',
+      'order_number' => '6',
+      'billing_profile' => $profile->id(),
+      'store_id' => $this->store->id(),
+    ]);
+    $order->save();
+    $this->order = $this->reloadEntity($order);
+  }
+
+  /**
+   * Tests the "needsRefresh" method and refresh settings.
+   *
+   * @group failing
+   */
+  public function testOrderCanRefresh() {
+    $refresher = $this->container->get('commerce_order.order_refresh');
+
+    $line_item = $this->lineItemStorage->createFromPurchasableEntity($this->variation2, [
+      'unit_price' => new Price('2.00', 'USD'),
+    ]);
+    $line_item->save();
+    $line_item = $this->reloadEntity($line_item);
+    $this->order->addLineItem($line_item);
+    $this->order->save();
+
+    $this->assertFalse($refresher->needsRefresh($this->order));
+
+    $order_type = OrderType::load($this->order->bundle());
+    $order_type->setRefreshFrequency(1)->save();
+
+    sleep(1);
+    $this->assertTrue($refresher->needsRefresh($this->order));
+
+    $order_type->setRefreshMode(OrderType::REFRESH_OWNER)->save();
+    $this->container->get('current_user')->setAccount($this->user);
+
+    sleep(1);
+    $this->assertFalse($refresher->needsRefresh($this->order));
+    $this->order->setOwner($this->user);
+    $this->assertTrue($refresher->needsRefresh($this->order));
+  }
+
+  /**
+   * Tests that a line item's title is updated based on product changes.
+   */
+  public function testLineItemTitleUpdate() {
+    $line_item = $this->lineItemStorage->createFromPurchasableEntity($this->variation2, [
+      'unit_price' => new Price('2.00', 'USD'),
+    ]);
+    $line_item->save();
+    $line_item = $this->reloadEntity($line_item);
+
+    $this->assertEquals($line_item->label(), 'Default testing product');
+    $this->order->addLineItem($line_item);
+    $this->order->save();
+
+    $this->variation2->getProduct()->setTitle('Changed title')->save();
+    $this->variation2->save();
+    $this->container->get('commerce_order.order_refresh')->refresh($this->order);
+
+    $this->variation2 = $this->reloadEntity($this->variation2);
+    $line_item = $this->reloadEntity($line_item);
+
+    $this->assertEquals($line_item->label(), 'Changed title');
+  }
+
+  /**
+   * Tests that a line item's unit price is set and updated on refresh.
+   */
+  public function testUnitPriceRefresh() {
+    /** @var \Drupal\commerce_order\Entity\LineItemInterface $line_item */
+    $line_item = $this->lineItemStorage->createFromPurchasableEntity($this->variation2);
+    $line_item->save();
+    $line_item = $this->reloadEntity($line_item);
+
+    $this->order->addLineItem($line_item);
+    $this->order->save();
+    $this->container->get('commerce_order.order_refresh')->refresh($this->order);
```

Comment:
Why do we use $this->container->get('commerce_order.order_refresh') all over the place in an OrderRefreshTest? Set a $this->orderRefresh at setUp().